apps/api/src/app/auth/services/auth.service.ts

```diff
@@ -89,20 +93,45 @@ export class AuthService {
         const withoutUsername = user.tokens.find(
           (i) => i.provider === AuthProviderEnum.GITHUB && !i.username && String(i.providerId) === String(profile.id)
         );
-
-        if (withoutUsername) {
-          await this.userRepository.update(
-            {
-              _id: user._id,
-              'tokens.providerId': profile.id,
-            },
-            {
-              $set: {
-                'tokens.$.username': profile.login,
+        // Checks if user's github primary email already exists but from a different authentication provider.
+        const withoutToken = user.tokens.findIndex((i) => i.provider === AuthProviderEnum.GITHUB) === -1;
+        const userUpdates: Promise<any>[] = [];
+
+        if (withoutUsername || email !== user.email) {
+          userUpdates.push(
+            this.userRepository.update(
+              {
+                _id: user._id,
+                'tokens.providerId': profile.id,
               },
-            }
+              {
+                $set: {
+                  ...(withoutUsername && { 'tokens.$.username': profile.login }),
+                  ...(email !== user.email && { email }),
+                },
+              }
+            )
           );
+        }
+        if (withoutToken) {
```

Comment:
Here we will update the credentials only if the user does not have a GitHub token?
Could there be a case when we would want to update the credentials we the user have GitHub token but it is not updated?

Reply:
@djabarovgeorge This was an existing logic, although I get the additional concern of updating an expired token (if it expires in the first place). Maybe @p-fernandez could help out here.

Reply:
Not entirely sure about that. @p-fernandez WDYT?